### PR TITLE
RD-2615 - Set default values for driver cores, memory and gc_threshold

### PR DESCRIPTION
**Pull request by nfsantos**

### CODE REVIEW

File: settings.sh [hunk 1]
```diff
@@ -54,6 +54,65 @@ then
...
+    if [ "$RAW_CACHE_GC_THRESHOLD" -ge 30 ]
+    then
+        RAW_CACHE_GC_THRESHOLD=30
+    fi
+    export RAW_CACHE_GC_THRESHOLD
+    printf  "\n: \${RAW_CACHE_GC_THRESHOLD:=\"%s\"}" "$RAW_CACHE_GC_THRESHOLD">> settings.local.sh
+fi
+
+
+
```
Comment (miguel-branco) [inline settings.sh:115]: What is the effect on an existing installation?

Comment (miguel-branco) [reply settings.sh:115]: The other option we can consider is creating a install.sh/install.py script that downloads this repo itself, checks the environment/dependencies, and then sets these values the first-time?

Otherwise, are there any effects in upgrading this repo?

Comment (nfsantos) [reply settings.sh:115]: The env variables for mem, core and cache size are written to the `settings.local.sh`, which is not in Git, so it will not be overwritten if the user does a `git pull`. 

They are generated only if they are not yet on `settings.local.sh` and are not set as env variables. So in the common case, the user downloads a new installation, and at first run these settings are added to `settings.local.sh` with values customized for the installation. In the following runs, it uses the values that were generated. The user can also manually edit the settings on `settings.local.sh` or set them as env variables, which takes precedence over auto-generating.

In short, the settings are auto-generated only after trying to load any local customization, as an alternative to loading them from `settings.default.sh`.
